include/algorithms/logistic_regression/logistic_regression_predict.h

```diff
@@ -193,8 +193,156 @@ class DAAL_EXPORT Batch : public classifier::prediction::Batch
+ */
+namespace interface2
+{
+/**
+ * <a name="DAAL-CLASS-ALGORITHMS__LOGISTIC_REGRESSION__PREDICTION__BATCHCONTAINER"></a>
+ * \brief Provides methods to run implementations of the logistic regression algorithm.
+ *        This class is associated with daal::algorithms::logistic_regression::prediction::interface1::Batch class
+ *        and supports method to compute logistic regression prediction
+ *
+ * \tparam algorithmFPType  Data type to use in intermediate computations, double or float
+ * \tparam method           logistic regression computation method, \ref Method
+ */
+template<typename algorithmFPType, Method method, CpuType cpu>
+class DAAL_EXPORT BatchContainer : public PredictionContainerIface
+{
+public:
+    /**
+     * Constructs a container for logistic regression model-based prediction with a specified environment
+     * \param[in] daalEnv   Environment object
+     */
+    BatchContainer(daal::services::Environment::env *daalEnv);
+    /** Default destructor */
+    ~BatchContainer();
+    /**
+     * Computes the result of logistic regression model-based prediction
+     * \return Status of computations
+     */
+    services::Status compute() DAAL_C11_OVERRIDE;
+};
+
+/**
+ * <a name="DAAL-CLASS-ALGORITHMS__LOGISTIC_REGRESSION__PREDICTION__BATCH"></a>
+ * \brief Predicts logistic regression results
+ * <!-- \n<a href="DAAL-REF-LOGISTIC-REGRESSION-ALGORITHM">logistic regression algorithm description and usage models</a> -->
+ *
+ * \tparam algorithmFPType  Data type to use in intermediate computations for the logistic regression algortithm, double or float
+ * \tparam method           logistic regression computation method, \ref Method
+ *
+ * \par Enumerations
+ *      - \ref Method                                       logistic regression prediction methods
+ *      - \ref classifier::prediction::NumericTableInputId  Identifiers of input Numeric Table objects
+ *                                                          for the logistic regression prediction algorithm
+ *      - \ref classifier::prediction::ModelInputId         Identifiers of input Model objects of the algorithm
+ *      - \ref classifier::prediction::ResultId             Identifiers of prediction results
+ *
+ * \par References
+ *      - \ref interface1::Model "Model" class
+ *      - \ref classifier::prediction::interface1::Input "classifier::prediction::Input" class
+ *      - \ref classifier::prediction::interface1::Result "classifier::prediction::Result" class
+ */
+template<typename algorithmFPType = DAAL_ALGORITHM_FP_TYPE, Method method = defaultDense>
+class DAAL_EXPORT Batch : public classifier::prediction::interface2::Batch
+{
+public:
+    typedef classifier::prediction::interface2::Batch super;
+
+    typedef algorithms::logistic_regression::prediction::Input     InputType;
+    typedef algorithms::classifier::interface2::Parameter          ParameterType;
+    typedef algorithms::classifier::prediction::interface2::Result ResultType;
```

Comment:
Static backward compatibility will be broken due to this change. I thought interface2 should be added for logistic regression Result to reduce changes related to static backward compatibility.